Two false negatives:
1. Darwin successfully builds and installs the binary but fails for some unknown reason:
```Done. Your build exited with 0.

/Users/travis/build.sh: line 109: shell_session_update: command not found
```
2. One of the instances tries to pull and build openjdk ?!
```
/tmp/nix-build-openjdk-8u122b00.drv-0/jdk8u-jdk8u122-b00/jdk/src/share/classes/com/sun/management/HotSpotDiagnosticMXBean.java:52: warning - Tag @see: reference not found: ManagementFactory#getPlatformMXBeans(Class)
1 warning
make: *** [/tmp/nix-build-openjdk-8u122b00.drv-0/jdk8u-jdk8u122-b00//make/Main.gmk:160: docs-only] Error 2
builder for â€˜/nix/store/wawhv7f072f79h8byniipfwfh81806w6-openjdk-8u122b00.drvâ€™ failed with exit code 2
```

--- a/pkgs/shells/rc/default.nix
+++ b/pkgs/shells/rc/default.nix
@@ -1,0 +1,42 @@
+{ stdenv, fetchurl, autoreconfHook
+, ncurses #acinclude.m4 wants headers for tgetent().
+, historySupport ? false
+, readline ? null
+}:
+
+stdenv.mkDerivation rec {
+  name = "rc-${version}";
+  version = "1.7.4";
+
+  src = fetchurl {
+    url = "http://static.tobold.org/rc/rc-${version}.tar.gz";
+    sha256 = "1n5zz6d6z4z6s3fwa0pscqqawy561k4xfnmi91i626hcvls67ljy";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+  buildInputs = [ ncurses ]
+    ++ stdenv.lib.optionals (readline != null) [ readline ];
+
+  configureFlags = [
+    "--enable-def-interp=${stdenv.shell}" #183
+    ] ++ stdenv.lib.optionals historySupport [ "--with-history" ]
+    ++ stdenv.lib.optionals (readline != null) [ "--with-edit=readline" ];
+
+  prePatch = ''
+    substituteInPlace configure.ac \
+      --replace "date -I" "echo 2015-05-13" #reproducible-build
+  '';
+
+  passthru = {
+    shellPath = "/bin/rc";
+  };
+
+  meta = with stdenv.lib; {
+    description = "The Plan 9 shell";
+    longDescription = "Byron Rakitzis' UNIX reimplementation of Tom Duff's Plan 9 shell.";
+    homepage = http://tobold.org/article/rc;
+    license = with licenses; zlib;
+    maintainers = with maintainers; [ ramkromberg ];
+    platforms = with platforms; all;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -3279,6 +3279,8 @@ in
 
   rawdog = callPackage ../applications/networking/feedreaders/rawdog { };
 
+  rc = callPackage ../shells/rc { };
+
   read-edid = callPackage ../os-specific/linux/read-edid { };
 
   redir = callPackage ../tools/networking/redir { };

